tests/progaudi.tarantool.tests/Pool/ConnectionPoolTests.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using ProGaudi.Tarantool.Client.Model;
+using ProGaudi.Tarantool.Client.Pool;
+using ProGaudi.Tarantool.Client.Tests.Pool.Mocks;
+using Xunit;
+using Shouldly;
+
+namespace ProGaudi.Tarantool.Client.Tests.Pool
+{
+    public class ConnectionPoolTests
+    {
+        [Fact]
+        public async Task PoolWithSingleInstance_Call_WorksCorrectly()
+        {
+            // Arrange
+            var nodes = new List<TarantoolNodeMockInfo>
+            {
+                new TarantoolNodeMockInfo { Alive = true, Uri = "host:12345", Uuid = Guid.NewGuid(), ReadOnly = false }
+            };
+            var boxEventRegistrator = new MockBoxEventRegistrator();
+            var (pool, _, _) = CreateConnectionPoolWithMocks(nodes, boxEventRegistrator);
+            var boxEvents = boxEventRegistrator.GetInstanceEvents(nodes[0].Uuid);
+            
+            // Act
+            await pool.Call("do_something", RequestMode.Any);
+
+            // Assert
+            Assert.Equal(PoolStatus.Connected, pool.Status);
+            boxEvents.Count.ShouldBe(1);
+            Assert.Equal("Call", boxEvents[0]);
+        }
+
+        [Theory]
+        [InlineData(RequestMode.Any, 2, 1, 1)]
+        [InlineData(RequestMode.Ro, 2, 0, 2)]
+        [InlineData(RequestMode.Rw, 2, 2, 0)]
+        [InlineData(RequestMode.PreferRo, 2, 0, 2)]
+        [InlineData(RequestMode.PreferRw, 2, 2, 0)]
+        public async Task PoolWithRoAndRw_AllRequestModes_WorksCorrectly(RequestMode requestMode, 
```

Comment:
Flaky test. For some reason sometimes this test fails with null-reference [here](https://github.com/tarantool/tarantool-csharp/pull/2/files#diff-ccbaf8d6d8808ff2c7bed0ee12d3bbcf2d76a9e608772190d02bfb3f38aa4913R27)